proto/xmtpv4/payer_api/payer_api.proto

```diff
@@ -0,0 +1,28 @@
+// Payer API
+syntax = "proto3";
+
+package xmtp.xmtpv4.payer_api;
+
+import "google/api/annotations.proto";
+import "xmtpv4/envelopes/envelopes.proto";
+
+option go_package = "github.com/xmtp/proto/v3/go/xmtpv4/payer_api";
+
+message PublishUnpaidEnvelopesRequest {
+  repeated xmtp.xmtpv4.envelopes.ClientEnvelope envelopes = 1;
+}
+
+message PublishUnpaidEnvelopesResponse {
+  repeated xmtp.xmtpv4.envelopes.OriginatorEnvelope originator_envelopes = 1;
+}
+
+// A narrowly scoped API for publishing messages through a payer
+service PayerApi {
+  // Publish envelope
+  rpc PublishUnpaidEnvelopes(PublishUnpaidEnvelopesRequest) returns (PublishUnpaidEnvelopesResponse) {
+    option (google.api.http) = {
+      post: "/mls/v2/payer/publish-envelopes"
```

Comment:
is this what we expect libxmtp to use?

Reply:
The `google.api.http` is for GRPC Gateway (which turns our GRPC Proto API into a HTTP/JSON API).

It would only be used in libxmtp in WASM environments, which don't have GRPC support